src/Interpreters/HashTablesStatistics.cpp

```diff
@@ -16,9 +16,14 @@ std::optional<HashTablesStatistics::Entry> HashTablesStatistics::getSizeHint(con
     if (!params.isCollectionAndUseEnabled())
         throw DB::Exception(DB::ErrorCodes::LOGICAL_ERROR, "Collection and use of the statistics should be enabled.");
 
-    std::lock_guard lock(mutex);
-    const auto cache = getHashTableStatsCache(params, lock);
-    if (const auto hint = cache->get(params.key))
+    Cache::MappedPtr hint;
+    {
+        std::lock_guard lock(mutex);
+        const auto cache = getHashTableStatsCache(params, lock);
+        hint = cache->get(params.key);
+    }
+
+    if (hint)
     {
         LOG_TRACE(
             getLogger("HashTablesStatistics"),
```

Comment:
BTW `getLogger("HashTablesStatistics")` grabs a mutex and performs lookup in a map.
It might make sense to lookup the logger only once and save it in HashTablesStatistics ctor

Reply:
Good point!